pytorch_lightning/plugins/ddp_sequential_plugin.py

```diff
@@ -0,0 +1,403 @@
+            raise MisconfigurationException(
+                f'The provided balance sum: {sum(self.balance)} does not'
+                f' match your Sequential length: {len(model.sequential_module)}')
+
+    def _skip_init_connections(self, trainer):
+        """
+        Skip initialization if torch is already initialized and we're in testing.
+        Returns: Whether to skip initialization
+
+        """
+        if torch_distrib.is_initialized() and trainer.testing:
+            return True
+        return False
+
+    def init_model_parallel_groups(self, trainer):
+        num_model_parallel = 1  # TODO currently no support for vertical model parallel
+        mpu.initialize_model_parallel(
+            model_parallel_size_=num_model_parallel,
+            pipeline_length=self.gpus_per_model
+        )
+
+    def _infer_check_num_gpus(self, trainer):
+        """
+        Infer the number of GPUs per model.
+        Args:
+            trainer: The trainer object.
+        Returns: The appropriate balance for the model
+        """
+        if isinstance(self.balance, list):
+            if len(self.balance) != trainer.world_size:
+                raise MisconfigurationException(
+                    "Pipe currently only supports splitting the module onto all available GPUs"
+                )
+            # User has defined a balance for his model
+            return len(self.balance)
+        # Assume that the user wants to balance his model on all GPUs
+        return trainer.world_size
+
+    def on_accelerator_exit_rpc_process(self, trainer) -> None:
+        if not trainer.testing:
+            torch_distrib.barrier()  # Ensure we await main process initialization
+
+            # Add trainer/configure_optimizers to the pipe model for access in all worker processes
+            rpc_pipe.PipeModel.trainer = trainer
+            del rpc_pipe.PipeModel.trainer.model.sequential_module
+            rpc_pipe.PipeModel.trainer.model.sequential_module = rpc_pipe.PipeModel
+            rpc_pipe.PipeModel.configure_optimizers = trainer.model.configure_optimizers
+        super().on_accelerator_exit_rpc_process(trainer)
+
+    def set_main_rpc_process(self):
+        self.main_rpc_process = torch_distrib.get_rank(group=mpu.get_pipeline_parallel_group()) == 0
+
+    def on_main_rpc_connection(self, trainer) -> None:
+        # Create pipe_module
+        model = trainer.get_model()
+        self._find_and_init_pipe_module(model)
+        if not trainer.testing:
+            torch_distrib.barrier()  # Ensure we join main process initialization
+            model.sequential_module.foreach_worker(register_optimizers, include_self=True)
+
+    def _check_arguments(self, trainer):
+        if trainer.amp_backend is not None:
+            raise MisconfigurationException(
+                'DDPSequentialPlugin is currently not supported in Automatic Mixed Precision')
+
+    def configure_ddp(
```

Comment:
```suggestion
    @staticmethod
    def configure_ddp(
```
these are not true class methods, right?
so not full instantiation is needed (so a little speedup)

Reply:
configure_ddp is not a staticmethod elsewhere. Let's keep it consistent. If needed we will make another PR to make more functions static ones.